packages/hardhat-core/src/internal/hardhat-network/provider/vm/rethnet.ts

```diff
@@ -117,11 +124,13 @@ export class RethnetAdapter implements VMAdapter {
     const trace = rethnetResult.trace!;
     for (const traceItem of trace) {
       if ("pc" in traceItem) {
-        await this._vmTracer.addStep(traceItem);
+        // TODO: these "as any" shouldn't be necessary, we had
+        // to add them after merging the changes in main
+        await this._vmTracer.addStep(traceItem as any);
       } else if ("executionResult" in traceItem) {
-        await this._vmTracer.addAfterMessage(traceItem);
+        await this._vmTracer.addAfterMessage(traceItem as any);
       } else {
-        await this._vmTracer.addBeforeMessage(traceItem);
+        await this._vmTracer.addBeforeMessage(traceItem as any);
```

Comment:
I think you are confusing the `VMDebugTracer`, which is the thing I removed, with the `VMTracer`, which is always used. I don't blame you, the term "tracing" is super overloaded here.

`VMTracer` is always used because the data it collects can be useful to build the stack trace and infer the error if the transaction reverts. (This is probably a mistake; we should actively re-trace reverted transactions instead of tracing everything in case it fails, but that's not something worth changing right now.)

`VMDebugTracer` is only meant to be used with `debug_*` methods.